### PR TITLE
e2e: Adds better log information during test runs

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Yesterday we had a failing e2e test run https://app.circleci.com/jobs/github/grafana/grafana/204720 but it's very hard to figure out what went wrong. This PR tries to remedy this see output from a local run below:

```log
(Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    3.7.0                                                                              │
  │ Browser:    Electron 73 (headless)                                                             │
  │ Specs:      1 found (smoketests.spec.ts)                                                       │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  smoketests.spec.ts                                                              (1 of 1)


  Smoke tests
Trying to login with: { username: 'admin', password: 'admin' }
Trying to login with: { username: 'admin', password: 'admin' }
Retrieving Selector: Username input field
Retrieving Selector: Password input field
Retrieving Selector: Login button
Logged in with { username: 'admin', password: 'admin' }
Adding data source with pluginName: TestData DB
Retrieving Selector: Data source plugin item TestData DB
Retrieving Selector: Data source settings page name input field
Retrieving Selector: Data source settings page name input field
Retrieving Selector: Data source settings page Save and Test button
Retrieving Selector: Data source settings page Alert
Retrieving Selector: Data source settings page Alert message
Added data source with name: e2e-1575964568708
Adding dashboard
Retrieving Selector: Dashboard navigation bar button Save dashboard
Retrieving Selector: Save dashboard title field
Retrieving Selector: Save dashboard title field
Retrieving Selector: Save dashboard button
Added dashboard with title: e2e-1575964568708
Retrieving Selector: Dashboard search item e2e-1575964568708
Retrieving Selector: Dashboard navigation bar button Add panel
Retrieving Selector: Add Panel Widget CTA Button Add Query
Retrieving Selector: Edit panel tab item Queries
Retrieving Selector: Test Data Query scenario select
Retrieving Selector: Edit panel tab item Visualization
Retrieving Selector: X-Axis section
Retrieving Selector: Dashboard navigation bar button Save dashboard
Retrieving Selector: Dashboard settings Save Dashboard Modal Save button
Retrieving Selector: Dashboard settings Go Back button
Retrieving Selector: Panel header title item Panel Title
Retrieving Selector: Panel header item Share
Retrieving Selector: Link to rendered image
    ✓ Login scenario, create test data source, dashboard, panel, and export scenario (8763ms)
Deleting data source with name: e2e-1575964568708
Deleting dashboard with uid: rmN47l-Zz


  1 passing (10s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     9 seconds                                                                        │
  │ Spec Ran:     smoketests.spec.ts                                                               │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  smoketests.spec.ts                       00:09        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:09        1        1        -        -        -

✨  Done in 23.64s.
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-e2e/package.json [hunk 1]
```diff
@@ -37,5 +36,8 @@
     "ts-loader": "6.2.1",
     "typescript": "3.7.2"
   },
-  "types": "src/index.ts"
+  "types": "src/index.ts",
+  "dependencies": {
+    "cypress": "3.7.0"
```
Comment (hugohaggmark) [inline packages/grafana-e2e/package.json]: I took the opportunity to upgrade the Cypress version too.

**Review by dprokop — APPROVED** [review]
Legit!

**Comment by mckn** [comment]
Looks really good! 🙌 Awesome job @hugohaggmark

File: packages/grafana-e2e/src/support/types.ts [hunk 2]
```diff
@@ -20,14 +20,19 @@ export const pageFactory = <S extends Selectors>({ url, selectors }: PageFactory
...
+      pageObjects[key] = () => {
+        e2e().logToConsole('Retrieving Selector:', value);
+        return e2e().get(Selector.fromAriaLabel(value));
+      };
     }
     if (typeof value === 'function') {
       // @ts-ignore
       pageObjects[key] = (text?: string) => {
         if (!text) {
+          e2e().logToConsole('Retrieving Selector:', value());
```
Comment (aknuds1) [inline packages/grafana-e2e/src/support/types.ts]: Is it better to evaluate `value()` just once here, by assigning the result to a variable? Just worried the result may change between the different calls.

File: packages/grafana-e2e/src/support/types.ts [hunk 3]
```diff
@@ -20,14 +20,19 @@ export const pageFactory = <S extends Selectors>({ url, selectors }: PageFactory
...
           return e2e().get(value());
         }
+        e2e().logToConsole('Retrieving Selector:', value(text));
```
Comment (aknuds1) [inline packages/grafana-e2e/src/support/types.ts]: Is it better to evaluate `value(text)` just once here, by assigning the result to a variable? Just worried the result may change between the different calls.

**Review by mckn — APPROVED** [review]

**Review by aknuds1 — COMMENTED** [review]
Looks good as far as I can tell, but have a couple of questions.

File: packages/grafana-e2e/src/support/types.ts (hunk 2 above)
Comment (hugohaggmark) [reply packages/grafana-e2e/src/support/types.ts]: Good catch, will do that!

File: packages/grafana-e2e/src/support/types.ts (hunk 3 above)
Comment (hugohaggmark) [reply packages/grafana-e2e/src/support/types.ts]: Good catch, will do that!

**Review by aknuds1 — APPROVED** [review]
LGTM!

**Pull request merged**